File: oldCode/test2.m
%Get Data from file
load testData.out;
format long;
S = length(testData);

%Generate the shift times.
Max = 300;
dt = 1e-10;
tv = zeros(Max,1);
tv(1) = dt;

for k = 1:Max-1
    tv(k+1) = tv(k) + (1e-10)*2^floor(k/10.0);
end

cv = zeros(size(tv));

%Start the Timer:
tic

for k = 1:length(tv)
    i = 1; j = 1; count = 0;
    t = tv(k);
    di = testData(1);
    dj = testData(1)+t;
    while (1)
        if (di < dj)
            i = i + 1;
            if (i > S) 
                break;
            end
            di = testData(i);
        elseif (di > dj)
            j = j + 1;
            if (j > S) 
                break;
            end
            dj = testData(j) + t;
        else
            count = count + 1;
            i = i + 1;
            if (i > S) 
                break;
            end
            di = testData(i);
        end
    end
    cv(k) = count;
end
toc

File: oldCode/test2.cpp
#include <iostream>  /* cout */
#include <string>   /* string */
#include <sstream>  /* istringstream, convert */
#include <fstream> /* ifstream */
#include <vector> /* vector */
#include <cmath> /* floor, pow */
#include <ctime>  /* clock */

using namespace std;

int main()  {
    
    //Read data from the file.
    string line;
    vector <double> data;
    double d;
    size_t count;
    ifstream infile;
    
    //convert data from string to double 
    infile.open("Data.out");
    while (infile.eof() == 0)  {
        getline (infile, line);
        istringstream convert(line);
        if ( !(convert >> d) ){
            cout << "Error in reading data!" << endl;
        }
        else data.push_back(d);
    }    
    infile.close();
    
    //Genearte the shift times.
    size_t d_size = data.size();
    int Max = 300;
    vector <double> times;
    times.push_back(1e-10);
    for (double k = 1; k < Max; k++)  {
        times.push_back(times.back() + (1e-10)*pow(2,floor(k/10.0) ) );
    }
    
    //Start the timer clock.
    clock_t start;
    start = clock();
    
    //Run the alg. for each start time.
    double di, dj, t;
    vector <int> cv;
    for ( int k = 0; k < Max; k++ )  {
        t = times[k];
        int i = 0;
        int j = 0;
        count = 0;
        di = data[0];
        dj = data[0]+t;
        while ( 1 )  {
            if (di < dj )  {
                if ( i++ >= d_size ) break;
                di = data[i];
            }
            else if (di > dj )  {
                if ( j++ >= d_size ) break;
                dj = data[j]+t;
            }
            else  {
                count += 1;
                if ( i++ >= d_size ) break;
                di = data[i];
            }
        }
        cv.push_back(count);
    }
    
    cout.precision(15);
    start = clock() - start;
    cout << ((float)start)/CLOCKS_PER_SEC << " seconds to run." << endl;    
}